### DISCUSSION TITLE
Is there a way to call mutate directly?

### DISCUSSION THREAD

**Discussion by ryantomaselli** [TanStack/query#1222] [Q&A]
Hey guys;

I would like to call mutate function returned by the useMutation hook multiple times. Of course this is a bad idea because each mutate function returned by useMutation has its own state (isLoading, isError etc.)

Does react-query expose something similar to apollo client's mutate function?
See https://www.apollographql.com/docs/react/api/core/ApolloClient/#ApolloClient.mutate

Cheers

**Comment by TkDodo** [comment]
Hmm, why can’t you call the same mutation function twice? Say you have two button, each with a click handler that fires off the mutation. 

If you need to track different loading states, I would call `useMutation` twice.

**Reply by ryantomaselli** [reply]
In my case the number of saves required is not known ahead of time. For example the user makes some changes to one or more items in a grid. My save will issue an xhr call for each item updated. If I "re-use" a mutation then the status of each individual save would be lost no?

**Reply by TkDodo** [reply]
oh okay, you can also just have one mutation that fires of multiple requests:

```
const [mutateThing] = useMutation(
    async (selection) => {
        return Promise.all(selection.map(id => updateThing(id))
    }
)

<button onClick={() => mutateThing(selection)}>Update</button>
```

you can `await` all calls separately or use [Promise.allsettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled), since `Promise.all` will immediately reject if one request fails, which is probably not what we want with mutations :)

**Reply by ryantomaselli** [reply]
Thanks for the responses @TkDodo. I am doing exactly as you state right now and it works great but my sense is that the various states returned by useMutation could not be used to know the state of the mutation. That said, I would need to confirm.
